Content.Shared/Anomaly/Effects/BluespaceAnomalySystem.cs

```diff
@@ -0,0 +1,87 @@
+using Content.Shared.Anomaly.Effects.Components;
+using Content.Shared.Mobs.Components;
+using Content.Shared.Teleportation.Components;
+using Robust.Shared.Map.Components;
+using Robust.Shared.Random;
+namespace Content.Shared.Anomaly.Effects;
+
+public sealed class BluespaceAnomalySystem : EntitySystem
+{
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly SharedTransformSystem _xform = default!;
+
+    /// <inheritdoc/>
+    public override void Initialize()
+    {
+        SubscribeLocalEvent<BluespaceAnomalyComponent, AnomalyPulseEvent>(OnPulse);
+        SubscribeLocalEvent<BluespaceAnomalyComponent, AnomalySupercriticalEvent>(OnSupercritical);
+        SubscribeLocalEvent<BluespaceAnomalyComponent, AnomalySeverityChangedEvent>(OnSeverityChanged);
+    }
+
+    private void OnPulse(EntityUid uid, BluespaceAnomalyComponent component, ref AnomalyPulseEvent args)
+    {
+        var xform = Transform(uid);
+        var range = component.MaxShuffleRadius * args.Severity;
+        var allEnts = _lookup.GetComponentsInRange<MobStateComponent>(xform.Coordinates, range)
+            .Select(x => x.Owner).ToList();
+        if (allEnts.Count % 2 == 0) //ensure even number
+            allEnts.RemoveAt(0);
+        allEnts.Add(uid);
+
+        var xformQuery = GetEntityQuery<TransformComponent>();
+        var transforms = new List<TransformComponent>();
+        foreach (var ent in allEnts)
+        {
+            if (xformQuery.TryGetComponent(ent, out var xf))
+                transforms.Add(xf);
+        }
+
+        var variation = component.MaxShuffleVariation * args.Severity;
+
+        // make sure we didn't pick up a transformless mob somehow
+        while (transforms.Any())
+        {
+            var t1 = _random.PickAndTake(transforms);
+            var t2 = _random.PickAndTake(transforms);
+            var c1 = t1.Coordinates.Offset(
+                new Vector2(_random.NextFloat(-variation, variation), _random.NextFloat(-variation, variation)));
+            var c2 = t2.Coordinates.Offset(
+                new Vector2(_random.NextFloat(-variation, variation), _random.NextFloat(-variation, variation)));
+
+            _xform.SetCoordinates(t1, c2);
+            _xform.SetCoordinates(t2, c1);
+        }
+    }
+
+    private void OnSupercritical(EntityUid uid, BluespaceAnomalyComponent component, ref AnomalySupercriticalEvent args)
+    {
+        var xform = Transform(uid);
+        var g = xform.GridUid;
+        if (g is not { } grid)
+            return;
+        if (!TryComp<MapGridComponent>(grid, out var gridComp))
+            return;
+        var gridBounds = gridComp.LocalAABB;
+
+        var xformQuery = GetEntityQuery<TransformComponent>();
+        foreach (var comp in _lookup.GetComponentsInRange<MobStateComponent>(xform.Coordinates, component.MaxShuffleRadius))
```

Comment:
how big is MaxShuffleRadius once it reaches this point normally?

Reply:
MaxShuffleRadius is somewhere between 0 and 10. It's a large area, but i think it's fastest to do it this way versus querying the component or something else.

Reply:
As long it is big enough to cover most of, lets say, boxstation. My only concern is that it doesnt become a weak meltdown compared to others since its so easy to escape the consequences of a random tp.

Reply:
my feeling was that due to it's near uncontrollable nature of teleporting around, having a not overly-dangerous supercritical event was better so that it wouldn't feel super unfair. At the very least, the teleport has a high likelihood of throwing you outside the station, which is very likely to be deadly.